src/test/java/com/google/sps/servlets/ProjectServletTest.java

```diff
@@ -0,0 +1,230 @@
+    when(request.getParameter("owners")).thenReturn("abc@xyz.com, xyz@abc.com");
+    servlet.doPost(request, response);
+    assertEquals(
+        6, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
+    writer.flush();
+    Entity projEntity = datastore.get(
+        KeyFactory.stringToKey(stringWriter.toString().replace("\"", "")));
+    assertEquals((String)projEntity.getProperty("name"), "MyProject8");
+    assertEquals((String)projEntity.getProperty("visibility"),
+                 DataUtils.PUBLIC);
+    assertEquals(
+        (ArrayList<String>)projEntity.getProperty("owners"),
+        new ArrayList<String>(Arrays.asList("abc@xyz.com", "xyz@abc.com")));
+    assertEquals(
+        (ArrayList<String>)projEntity.getProperty("editors"),
+        new ArrayList<String>(Arrays.asList("aaa@bbb.com", "bbb@ccc.com")));
+  }
+
+  @Test
+  public void update() throws IOException, EntityNotFoundException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(
+        5, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
+    when(request.getParameter("mode")).thenReturn("update");
+    when(request.getParameter("proj-id")).thenReturn(projId);
+    when(request.getParameter("proj-name")).thenReturn("MyProject8");
+    when(request.getParameter("visibility")).thenReturn(DataUtils.PUBLIC);
+    when(request.getParameter("editors"))
+        .thenReturn("aaa@bbb.com, bbb@ccc.com");
+    when(request.getParameter("owners")).thenReturn("abc@xyz.com, xyz@abc.com");
+    servlet.doPost(request, response);
+    assertEquals(
+        5, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
+    writer.flush();
+    Entity projEntity = datastore.get(KeyFactory.stringToKey(projId));
+    assertEquals((String)projEntity.getProperty("name"), "MyProject8");
+    assertEquals((String)projEntity.getProperty("visibility"),
+                 DataUtils.PUBLIC);
+    assertEquals(
+        (ArrayList<String>)projEntity.getProperty("owners"),
+        new ArrayList<String>(Arrays.asList("abc@xyz.com", "xyz@abc.com")));
+    assertEquals(
+        (ArrayList<String>)projEntity.getProperty("editors"),
+        new ArrayList<String>(Arrays.asList("aaa@bbb.com", "bbb@ccc.com")));
+    assertFalse(((String)projEntity.getProperty("name"))
+                    .equals("2020-08-12T05:39:02.383Z"));
+  }
+
+  @Test
+  public void delete() throws IOException, EntityNotFoundException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(
+        5, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
+    when(request.getParameter("mode")).thenReturn("update");
+    when(request.getParameter("proj-id")).thenReturn(projId);
+    when(request.getParameter("delete")).thenReturn("true");
+    servlet.doPost(request, response);
+    assertEquals(
+        4, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
```

Comment:
Can you also test requesting the specific project id and ensure it doesn't return it?